woocommerce/google-listings-and-ads
# Contact info notification and issue

### Changes proposed in this Pull Request:

This PR checks if contact information has been set for users that already went through the onboarding process.
If it hasn't been set it prompts them to do so with an Inbox Notifcation and an entry added to the Issues table.

Closes #910

### Detailed test instructions:
1. Edit `gla_db_version` in the options table and set it to an older version (like 1.2.1), to trigger the DB table to update
2. Confirm we already completed Merchant Center setup
3. Send a POST request to `/wc/gla/mc/contact-information` containing a valid phone number (will also update store address)
4. On the connection test page clear the transient for the Merchant Statuses
5. Reload the Product Feed page and confirm there is no issue showing up for updating contact info
6. Go to WooCommerce > Home and confirm there is no notice prompting to update the contact info
7. Send a POST request to `/wc/gla/mc/contact-information` with the `phone_number` set to `null`
8. On the connection test page clear the transient for the Merchant Statuses
9. Delete the option `gla_contact_info_setup` in the `wp_options` table
10. Reload the Product Feed page and confirm there is an issue showing up for updating contact info
![image](https://user-images.githubusercontent.com/11388669/127648562-39263bfc-329d-41fc-a230-295e9c5ea6ce.png)
11. Check the issue link and confirm it's pointing to the full link `...?page=wc-admin&path=/google/settings&subpath=/edit-contact-information` (it was previously limited to 100 characters in the DB table)
12. Go to WooCommerce > Home and confirm there is a notice prompting to update the contact info
![image](https://user-images.githubusercontent.com/11388669/127648329-9377b7fe-291d-4003-8d10-70a72a5cb7cc.png)
13. Remove the previous note from the DB table `wp_wc_admin_notes` with name `gla-contact-information`
14. Reload WooCommerce > Home and confirm the notice is gone
15. Update the address in the Merchant Center dashboard and make it different to the one stored in WC settings
16. Manually clear the transient `mc_contact_info` in the `wp_options` table
17. Reload WooCommerce > Home and confirm the notice appears again

### Changelog entry

> Add - Prompt users to setup their contact information.
